src/runtimes/node/native_modules/plugin.js

```diff
@@ -0,0 +1,68 @@
+const path = require('path')
+
+const readPackageJson = require('read-package-json-fast')
+
+const { isNativeModule } = require('./detector')
+
+const packageName = /^([^@][^/]*|@[^/]*\/[^/]+)(?:\/|$)/
+
+const findNativeModule = (packageJsonPath, cache) => {
+  if (cache[packageJsonPath] === undefined) {
+    // eslint-disable-next-line fp/no-mutation, no-param-reassign, promise/prefer-await-to-then
+    cache[packageJsonPath] = readPackageJson(packageJsonPath).then(
+      (data) => Boolean(isNativeModule(data)),
+      () => {},
+    )
+  }
+
+  return cache[packageJsonPath]
+}
+
+const externalNativeModulesPlugin = (externalizedModules) => ({
+  name: 'external-native-modules',
+  setup(build) {
+    const cache = {}
+
+    // eslint-disable-next-line complexity, max-statements
+    build.onResolve({ filter: /^([^./]*)$/ }, async (args) => {
+      const package = packageName.exec(args.path)
+
+      if (!package) return
+
+      // eslint-disable-next-line fp/no-let
+      let directory = args.resolveDir
+
+      // eslint-disable-next-line fp/no-loops
+      while (true) {
```

Comment:
[sand] Should we put the stop condition (`parentDirectory !== directory`) here?

Reply:
That means having `parentDirectory` outside the scope of the loop and initialising them both together, which feels a bit odd? What do you think?

Reply:
Good point. Since `directory` is already in the outer scope we could be consistent with `parentDirectory` too.
I'm good either way.

Reply:
It's just that placing `parentDirectory` in the outer scope means duplicating the `path.dirname` assignment: once outside, to satisfy the loop condition, and a second time inside to calculate the parent directory of each iteration, unless I'm missing what you're suggesting.

Reply:
Yes, you're right. We can keep it as is 👍